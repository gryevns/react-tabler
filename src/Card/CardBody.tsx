import React from "react";
import cn from "classnames";

export interface CardBodyProps {
    className?: string;
    children?: React.ReactNode;
    loading?: boolean;
    style?: React.CSSProperties;
}

const Loading = ({ children }) => (
    <div className="dimmer active">
        <div className="loader"></div>
        <div className="dimmer-content">{children}</div>
    </div>
);

function CardBody({ className, children, style, loading = false }: CardBodyProps) {
    const classes = cn("card-body", className);
    return (
        <div className={classes} style={style}>
            {loading ? <Loading>{children}</Loading> : children}
        </div>
    );
}

export default CardBody;
