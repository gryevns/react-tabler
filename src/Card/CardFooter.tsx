import React from "react";
import cn from "classnames";

export interface CardFooterProps {
    className?: string;
    children: React.ReactNode;
}

function CardFooter({ className, children }: CardFooterProps) {
    const classes = cn("card-footer", className);
    return <div className={classes}>{children}</div>;
}

export default CardFooter;
