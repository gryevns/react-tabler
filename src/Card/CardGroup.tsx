import React from "react";
import cn from "classnames";

export interface CardGroupProps {
    className?: string;
    children?: React.ReactNode;
}

function CardGroup({ className, children }: CardGroupProps) {
    const classes = cn("card-group", className);
    return <div className={classes}>{children}</div>;
}

export default CardGroup;
