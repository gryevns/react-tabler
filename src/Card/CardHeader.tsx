import * as React from "react";
import cn from "classnames";

export interface CardHeaderProps {
    className?: string;
    children: React.ReactNode;
}

function CardHeader({ className, children }: CardHeaderProps) {
    const classes = cn("card-header", className);
    return <div className={classes}>{children}</div>;
}

export default CardHeader;
