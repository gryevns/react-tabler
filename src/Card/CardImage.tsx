import React from "react";
import cn from "classnames";

export interface CardImageProps extends React.ImgHTMLAttributes<HTMLElement> {
    className?: string;
    top?: boolean;
    bottom?: boolean;
}

const CardImage = ({ className, top, bottom, ...props }: CardImageProps) => {
    const classes = cn(
        { "card-img-bottom": bottom, "card-img-top": top },
        className,
    );
    return <img className={classes} {...props} />;
};

export default CardImage;
