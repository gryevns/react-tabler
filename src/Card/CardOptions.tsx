import * as React from "react";
import cn from "classnames";

export interface CardOptionsProps {
    className?: string;
    children: React.ReactNode;
}

function CardOptions({ className, children }: CardOptionsProps) {
    const classes = cn("card-options", className);
    return <div className={classes}>{children}</div>;
}

export default CardOptions;
