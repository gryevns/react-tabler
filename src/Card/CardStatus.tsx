import * as React from "react";
import cn from "classnames";
import { Color } from "../common";

export interface CardStatusProps {
    color: Color;
    top?: boolean;
    right?: boolean;
    bottom?: boolean;
    left?: boolean;
}

function CardStatus({ color, top, right, bottom, left }: CardStatusProps) {
    const classes = cn({
        "card-status": true,
        [`bg-${color}`]: true,
        "card-status-top": top,
        "card-status-right": right,
        "card-status-bottom": bottom,
        "card-status-left": left,
    });
    return <div className={classes} />;
}

export default CardStatus;
