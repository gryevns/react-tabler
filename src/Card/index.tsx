import React from "react";
import cn from "classnames";
import CardOptions from "./CardOptions";
import CardStatus from "./CardStatus";
import CardHeader from "./CardHeader";
import CardTitle from "./CardTitle";
import CardBody from "./CardBody";
import CardFooter from "./CardFooter";

export type Size = "sm" | "md" | "lg";

export interface CardProps {
    className?: string;
    children?: React.ReactNode;
    stacked?: boolean;
    aside?: boolean;
    size?: Size;
    style?: React.CSSProperties;
}

const Card = function ({
    className,
    children,
    aside,
    stacked,
    size,
    style,
}: CardProps) {
    const classes = cn(
        {
            card: true,
            aside: aside,
            [`card-${size}`]: size,
            "card-stacked": stacked,
        },
        className,
    );

    return (
        <div className={classes} style={style}>
            {children}
        </div>
    );
};

Card.Header = CardHeader;
Card.Title = CardTitle;
Card.Body = CardBody;
Card.Footer = CardFooter;
Card.Options = CardOptions;
Card.Status = CardStatus;

export default Card;
