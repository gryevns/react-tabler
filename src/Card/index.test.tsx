import React from "react";
import { shallow } from "enzyme";

import Card from "./index";

describe("Card", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(
            <Card>
                <Card.Header>
                    <Card.Title>Card title</Card.Title>
                </Card.Header>
                <Card.Body>
                    <p>This is some text within a card body.</p>
                </Card.Body>
            </Card>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should match size snapshot", () => {
        ["sm" as const, undefined, "md" as const, "lg" as const].map(size => {
            const wrapper = shallow(
                <Card size={size}>
                    <Card.Body>This is some text within a card body.</Card.Body>
                </Card>,
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should match active snapshot", () => {
        [true, false].map(active => {
            const wrapper = shallow(
                <Card active={active}>
                    <Card.Body>Test</Card.Body>
                </Card>,
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should match group snapshot", () => {
        const wrapper = shallow(
            <Card.Group>
                <Card inactive>
                    <Card.Body>First card</Card.Body>
                </Card>
                <Card active>
                    <Card.Body>Second card</Card.Body>
                </Card>
            </Card.Group>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should match status snaphost", () => {
        ["top", "bottom", "left"].map(position => {
            const props = { [position]: true, color: "purple" as const };
            const wrapper = shallow(<Card.Status {...props} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
    it("should match stacked snapshot", () => {
        const wrapper = shallow(<Card stacked />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should match loading snapshot", () => {
        const wrapper = shallow(<Card.Body loading={true} />);
        expect(wrapper).toMatchSnapshot();
    });
});
