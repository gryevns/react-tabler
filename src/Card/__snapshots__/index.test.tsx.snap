// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match active snapshot 1`] = `
<div
  className="card card-active"
>
  <CardBody>
    Test
  </CardBody>
</div>
`;

exports[`Card should match active snapshot 2`] = `
<div
  className="card"
>
  <CardBody>
    Test
  </CardBody>
</div>
`;

exports[`Card should match group snapshot 1`] = `
<div
  className="card-group"
>
  <Component
    inactive={true}
  >
    <CardBody>
      First card
    </CardBody>
  </Component>
  <Component
    active={true}
  >
    <CardBody>
      Second card
    </CardBody>
  </Component>
</div>
`;

exports[`Card should match loading snapshot 1`] = `
<div
  className="card-body"
>
  <Loading />
</div>
`;

exports[`Card should match size snapshot 1`] = `
<div
  className="card card-sm"
>
  <CardBody>
    This is some text within a card body.
  </CardBody>
</div>
`;

exports[`Card should match size snapshot 2`] = `
<div
  className="card"
>
  <CardBody>
    This is some text within a card body.
  </CardBody>
</div>
`;

exports[`Card should match size snapshot 3`] = `
<div
  className="card card-md"
>
  <CardBody>
    This is some text within a card body.
  </CardBody>
</div>
`;

exports[`Card should match size snapshot 4`] = `
<div
  className="card card-lg"
>
  <CardBody>
    This is some text within a card body.
  </CardBody>
</div>
`;

exports[`Card should match snapshot 1`] = `
<div
  className="card"
>
  <CardHeader>
    <Component>
      Card title
    </Component>
  </CardHeader>
  <CardBody>
    <p>
      This is some text within a card body.
    </p>
  </CardBody>
</div>
`;

exports[`Card should match stacked snapshot 1`] = `
<div
  className="card card-stacked"
/>
`;

exports[`Card should match status snaphost 1`] = `
<div
  className="card-status bg-purple card-status-top"
/>
`;

exports[`Card should match status snaphost 2`] = `
<div
  className="card-status bg-purple card-status-bottom"
/>
`;

exports[`Card should match status snaphost 3`] = `
<div
  className="card-status bg-purple card-status-left"
/>
`;
